pkg/didcomm/protocol/exchange/state/state.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	//NullState is the did-exchange protocol initial state
+	NullState = "null"
+	//InvitedState is the did-exchange protocol invited state
+	InvitedState = "invited"
+	//RequestedState is the did-exchange protocol requested state
+	RequestedState = "requested"
+	//RespondedState is the did-exchange protocol responded state
+	RespondedState = "responded"
+	//CompleteState is the did-exchange protocol completed state
+	CompleteState = "completed"
+
+	//Reqquest Types supported in Did-exchange
+	invitation = "invitation"
+	response   = "response"
+	request    = "request"
+	ack        = "ack"
+)
+
+//Context defines the DIDComm state context
+type Context struct {
+	Store storage.Store
+	State *State
+}
+
+// New creates the new context
+func New(store storage.Store, state *State) *Context {
+	return &Context{
+		Store: store,
+		State: state,
+	}
+}
+
+// DidCommState describes the internal State machine for Inviter and Invitee. Reference: https://github.com/hyperledger/aries-rfcs/blob/master/features/0023-did-exchange/chrome_2019-01-29_07-59-38.png
+type DidCommState interface {
```

Comment:
Sounds like this should be renamed to some short variant of "DidExchangeStateMachine".

Reply:
@llorllale I would have assumed the StateMachine was part of the service subpackage.

(presumably the StateMachine object isn't accessed by the normal developer?)

Reply:
shall i move this to service subpackage ?

Reply:
I would personally move it to an `exchange` package, whether it's the exported one or an `/internal/` one as @troyronda seems to suggest.

Reply:
My point is that the top level exchange package (and other top level protocol packages) should only contain developer facing APIs. Keep the GoDocs clean.

Reply:
Also - I don’t think StateMachine will need to be accessed outside the Exchange protocol package and sub packages, right?